cli/src/commands/generate/components/node.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+
+import { initLogger } from '../../helper.js';
+import { SchemaDirectory } from '../schema-directory.js';
+import { mergeSchemas } from '../util.js';
+import { getPropertyValue } from './property.js';
+
+/**
+ * Instantiate an individual node from its definition, resolving $refs if appropriate. 
+ * @param nodeDef The definition of the node.
+ * @param schemaDirectory The schema directory to resolve refs against.
+ * @param debug Whether to log debug detail.
+ * @returns An instantiated node.
+ */
+export function instantiateNode(nodeDef: any, schemaDirectory: SchemaDirectory, debug: boolean = false): any {
```

Comment:
Is unknown any better than any? Is that not just going to cause failed type checks later on if we try to introduce any types?

Reply:
it depends on how you're using it. If you don't plan on extracting any info from the thing, then unknown is definitely better because you're telling the type checker that you truly don't know what's in it and to let you know if you try to access it incorrectly. In cases where you are accessing it for something, then unknown might not be the best, but there are almost always better alternatives to `any`